script.js

```diff
@@ -1,4 +1,456 @@
+var gameStatus = "mode selection";
+var gameMode = "RANK";
+var concatenateNum = "";
+var rollDiceResult = [];
+var numOfDice = 2;
+var leaderBoardResults = "";
+var gameVersion = "";
+var numOfPlayers = 2;
+var playersArray = [];
+var turnCounter = 0;
+var knockoutTurn = "start knockout";
+var tempWinner = {};
+
+var player1 = {};
+var player2 = {};
+var nextPlayer = {};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "";
+
+  var inputValidationResults = inputValidation(input);
+
+  if (inputValidationResults == true) {
+    if (gameStatus == "mode selection") {
+      gameStatus = "version selection";
+
+      if (input.toUpperCase() == "KNOCKOUT") {
+        gameMode = "KNOCKOUT";
+      } else {
+        gameMode = "RANK";
+      }
+
+      myOutputValue = `You have selected the ${input} mode.<br><br>
+      Please select the version that you would like to play by entering
+        "Normal" or "Lowest Combined"`;
+
+      return myOutputValue;
+    } else if (gameStatus == "version selection") {
+      gameStatus = "player selection";
+
+      if (input.toUpperCase() == "LOWEST COMBINED") {
+        gameVersion = "lowestCombined";
+      } else {
+        gameVersion = "normal";
+      }
+
+      myOutputValue = `You have selected the ${input} version.<br><br>
+      Please enter the number of players.`;
+
+      return myOutputValue;
+    } else if (gameStatus == "player selection") {
+      numOfPlayers = input;
+      createPlayersArray(input);
+      gameStatus = "dice selection";
+
+      myOutputValue = `You have selected ${input} players.<br><br>
+      Please enter the number of dice to play.`;
+
+      return myOutputValue;
+    } else if (gameStatus == "dice selection") {
+      numOfDice = input;
+      gameStatus = "game intro";
+
+      myOutputValue = `You have selected ${input} dice.<br><br>
+      Please click submit to start rolling!`;
+
+      return myOutputValue;
+    } else if (gameStatus == "game intro") {
+      if (gameMode == "RANK") {
+        return playGame(numOfPlayers);
+      } else if (gameMode == "KNOCKOUT") {
+        return playKnockoutGame(numOfPlayers);
+      }
+    }
+  } else {
+    return inputValidationResults;
+  }
+};
+
+// Function for input validation
+
+var inputValidation = function (input) {
+  var returnStatement = "";
+  if (gameStatus == "version selection") {
+    if (
+      !(
+        input.toUpperCase() == "NORMAL" ||
+        input.toUpperCase() == "LOWEST COMBINED"
+      )
+    ) {
+      returnStatement = `Wrong input. Please enter only "Normal" or "Lowest Combined" to start the game.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
```

Comment:
Great input validation as a helper function, using modes to combine all input validation moments into a single function!